From 5bf714b008210c88dfa9cde6b921ebdc0e562664 Mon Sep 17 00:00:00 2001
From: Andrew Borodin <aborodin@vmail.ru>
Date: Thu, 15 Oct 2020 15:16:24 +0300
Subject: [PATCH] Ticket #4131: use G_UNICODE_SPACING_MARK instead of
 G_UNICODE_COMBINING_MARK.

G_UNICODE_COMBINING_MARK is deperecated in glib >= 30.

Signed-off-by: Andrew Borodin <aborodin@vmail.ru>
---
 lib/strutil/strutilutf8.c |  2 +-
 src/viewer/ascii.c        | 10 ++--------
 2 files changed, 3 insertions(+), 9 deletions(-)

diff --git a/lib/strutil/strutilutf8.c b/lib/strutil/strutilutf8.c
index efec6be54..5ac0015e6 100644
--- a/lib/strutil/strutilutf8.c
+++ b/lib/strutil/strutilutf8.c
@@ -70,7 +70,7 @@ str_unichar_iscombiningmark (gunichar uni)
     GUnicodeType type;
 
     type = g_unichar_type (uni);
-    return (type == G_UNICODE_COMBINING_MARK)
+    return (type == G_UNICODE_SPACING_MARK)
         || (type == G_UNICODE_ENCLOSING_MARK) || (type == G_UNICODE_NON_SPACING_MARK);
 }
 
diff --git a/src/viewer/ascii.c b/src/viewer/ascii.c
index 67b653434..e2a3ce934 100644
--- a/src/viewer/ascii.c
+++ b/src/viewer/ascii.c
@@ -162,12 +162,6 @@
 
 /*** file scope macro definitions ****************************************************************/
 
-#if GLIB_CHECK_VERSION (2, 30, 0)
-#define SPACING_MARK G_UNICODE_SPACING_MARK
-#else
-#define SPACING_MARK G_UNICODE_COMBINING_MARK
-#endif
-
 /* The Unicode standard recommends that lonely combining characters are printed over a dotted
  * circle. If the terminal is not UTF-8, this will be replaced by a dot anyway. */
 #define BASE_CHARACTER_FOR_LONELY_COMBINING 0x25CC      /* dotted circle */
@@ -266,7 +260,7 @@ mcview_is_spacing_mark (const WView * view, int c)
 {
 #ifdef HAVE_CHARSET
     if (view->utf8)
-        return g_unichar_type (c) == SPACING_MARK;
+        return g_unichar_type (c) == G_UNICODE_SPACING_MARK;
 #else
     (void) view;
     (void) c;
@@ -535,7 +529,7 @@ mcview_next_combining_char_sequence (WView * view, mcview_state_machine_t * stat
             return i;
         if (!mcview_ismark (view, cs[i]) || !mcview_isprint (view, cs[i]))
             return i;
-        if (g_unichar_type (cs[i]) == SPACING_MARK)
+        if (g_unichar_type (cs[i]) == G_UNICODE_SPACING_MARK)
         {
             /* Only allow as the first combining char. Stop processing in either case. */
             if (i == 1)
-- 
2.24.1

